ably/realtime_channel.go

```diff
@@ -615,6 +703,34 @@ func (c *RealtimeChannel) isActive() bool {
 	return c.state == ChannelStateAttaching || c.state == ChannelStateAttached
 }
 
+func (c *RealtimeChannel) channelOpts() *channelOptions {
+	return c.options
+}
+
+func (c *RealtimeChannel) setParams(params ChannelParams) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	c.params = params
+}
+
+func (c *RealtimeChannel) setModes(modes []ChannelMode) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	c.modes = modes
+}
+
+func (c *RealtimeChannel) Modes() []ChannelMode {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	return c.modes
```

Comment:
This won't be read-only. This should return a copy of the original slice.

Reply:
resolved